### PR TITLE
fix(bugs): fixed a log of bugs

### LINKED ISSUE #643: replica's binlog would be reset, once pod is reconstructed.
https://github.com/presslabs/mysql-operator/blob/47436cc5edd3d974239e45c994ccf7e22afa65e0/pkg/sidecar/appconf.go#L196

### LINKED ISSUE #692: When the mysqlConf is updated and the master node becomes the slave node, the node fails.
The two clusters present different problems. The new master node is OK.
```
# kubectl get mysql -nkme-mysql-10272    
NAME      READY   REPLICAS   AGE
lzx1      False   2          145m
lzx2      False   2          145m

# kubectl get pod -nkme-mysql-10272  
NAME              READY   STATUS    RESTARTS   AGE
lzx1-mysql-0      4/4     Running   0          52m
lzx1-mysql-1      4/4     Running   0          83m
lzx2-mysql-0      4/4     Running   0          52m
lzx2-mysql-1      4/4     Running   0          83m
```

lzx1 lzx1-mysql-0 log:
```
2021-06-04T10:11:31.028529Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.20-11) starting as process 1
2021-06-04T10:11:31.041698Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2021-06-04T10:11:31.771601Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2021-06-04T10:11:32.125889Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: '/var/lib/mysql/mysqlx.sock' bind-address: '::' port: 33060
2021-06-04T10:11:32.250403Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2021-06-04T10:11:32.354015Z 0 [Warning] [MY-010604] [Repl] Neither --relay-log nor --relay-log-index were used; so replication may break when this MySQL server acts as a slave and has his hostname changed!! Please use '--relay-log=lzx1-mysql-0-relay-bin' to avoid this problem.
2021-06-04T10:11:32.357097Z 0 [Warning] [MY-010539] [Repl] Recovery from master pos 175196 and file mysql-bin.000001 for channel ''. Previous relay log pos and relay log file had been set to 61754, ./lzx1-mysql-0-relay-bin.000002 respectively.
2021-06-04T10:11:32.392419Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.20-11'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 11, Revision 5b5a5d2.
2021-06-04T10:11:59.189718Z 33 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='lzx1-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='mysql-bin.000001', master_log_pos= 175196, master_bind=''. New state master_host='lzx1-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
2021-06-04T10:11:59.282748Z 35 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
2021-06-04T10:11:59.286711Z 35 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@lzx1-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
2021-06-04T10:11:59.289104Z 35 [ERROR] [MY-010557] [Repl] Error reading packet from server for channel '': Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been (server_errno=1236)
2021-06-04T10:11:59.289138Z 35 [ERROR] [MY-013114] [Repl] Slave I/O for channel '': Got fatal error 1236 from master when reading data from binary log: 'Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been', Error_code: MY-013114
2021-06-04T10:12:50.966490Z 77 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
2021-06-04T10:12:50.968930Z 77 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@lzx1-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
2021-06-04T10:12:50.971123Z 77 [ERROR] [MY-010557] [Repl] Error reading packet from server for channel '': Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been (server_errno=1236)
2021-06-04T10:12:50.971166Z 77 [ERROR] [MY-013114] [Repl] Slave I/O for channel '': Got fatal error 1236 from master when reading data from binary log: 'Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been', Error_code: MY-013114
```

lzx2 log:
```
2021-06-04T11:05:29.381914Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.20-11) starting as process 1
2021-06-04T11:05:29.393882Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2021-06-04T11:05:30.008040Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2021-06-04T11:05:30.274956Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: '/var/lib/mysql/mysqlx.sock' bind-address: '::' port: 33060
2021-06-04T11:05:30.393645Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2021-06-04T11:05:30.474279Z 0 [Warning] [MY-010604] [Repl] Neither --relay-log nor --relay-log-index were used; so replication may break when this MySQL server acts as a slave and has his hostname changed!! Please use '--relay-log=lzx2-mysql-0-relay-bin' to avoid this problem.
2021-06-04T11:05:30.479093Z 0 [Warning] [MY-010539] [Repl] Recovery from master pos 1082920 and file mysql-bin.000001 for channel ''. Previous relay log pos and relay log file had been set to 1070, ./lzx2-mysql-0-relay-bin.000002 respectively.
2021-06-04T11:05:30.535846Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.20-11'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 11, Revision 5b5a5d2.
2021-06-04T11:05:36.914128Z 22 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='lzx2-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='mysql-bin.000001', master_log_pos= 1082920, master_bind=''. New state master_host='lzx2-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
2021-06-04T11:05:59.217593Z 44 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='lzx2-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''. New state master_host='lzx2-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
2021-06-04T11:05:59.225106Z 46 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
2021-06-04T11:05:59.228631Z 46 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@lzx2-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
2021-06-04T11:05:59.238725Z 47 [ERROR] [MY-010584] [Repl] Slave SQL for channel '': Could not execute Update_rows event on table sys_operator.heartbeat; Can't find record in 'heartbeat', Error_code: 1032; handler error HA_ERR_KEY_NOT_FOUND; the event's master log mysql-bin.000001, end_log_pos 1168, Error_code: MY-001032
2021-06-04T11:05:59.238754Z 47 [Warning] [MY-010584] [Repl] Slave: Can't find record in 'heartbeat' Error_code: MY-001032
2021-06-04T11:05:59.238768Z 47 [ERROR] [MY-010586] [Repl] Error running query, slave SQL thread aborted. Fix the problem, and restart the slave SQL thread with "SLAVE START". We stopped at log 'mysql-bin.000001' position 808
2021-06-04T11:18:50.117442Z 694 [ERROR] [MY-010584] [Repl] Slave SQL for channel '': Could not execute Update_rows event on table sys_operator.heartbeat; Can't find record in 'heartbeat', Error_code: 1032; handler error HA_ERR_KEY_NOT_FOUND; the event's master log mysql-bin.000001, end_log_pos 1168, Error_code: MY-001032
2021-06-04T11:18:50.117468Z 694 [Warning] [MY-010584] [Repl] Slave: Can't find record in 'heartbeat' Error_code: MY-001032
2021-06-04T11:18:50.117477Z 694 [ERROR] [MY-010586] [Repl] Error running query, slave SQL thread aborted. Fix the problem, and restart the slave SQL thread with "SLAVE START". We stopped at log 'mysql-bin.000001' position 808
```

### PR DESCRIPTION
**Pull request by cndoit18**
Signed-off-by: cndoit18 <cndoit18@outlook.com>

Too much, I need to find Issues for it.
Can I help me together? please

fixes #643
fixes #605
fixes #692

---
 - [x] I've made sure the [Changelog.md](https://github.com/presslabs/mysql-operator/blob/master/Changelog.md) will remain up-to-date after this PR is merged.

### CODE REVIEW

File: charts/mysql-operator/templates/statefulset.yaml
Comment (AMecea) [inline charts/mysql-operator/templates/statefulset.yaml:52]: Why pointing to the orchestrator service and not leting to connect to the local orchestrator instance?

File: pkg/sidecar/configs.go
Comment (AMecea) [inline pkg/sidecar/configs.go:155]: I think you should use the `getOrdinalFromHostname()` function, much more now when you made it better. 😄

**Review by AMecea — COMMENTED** [review]
Hi @cndoit18, Thank you very much for your contribution! It looks good. I have only one question and a suggestion.

Sorry for the delay, I was caught in other projects. 😞

File: pkg/sidecar/configs.go
Comment (cndoit18) [reply pkg/sidecar/configs.go:155]: I just think it would improve the performance a bit if it is judged directly, it is not too important. Using `getOrdinalFromHostname` would also work

File: charts/mysql-operator/templates/statefulset.yaml
Comment (cndoit18) [reply charts/mysql-operator/templates/statefulset.yaml:52]: https://github.com/presslabs/mysql-operator/blob/master/pkg/controller/mysqlcluster/internal/syncer/statefullset.go#L324-L327

Because the sidecar in MySQL will use this API (e.g. in preStop), if you use localhost directly, these components will not be accessible

**Comment by cndoit18** [comment]
> Hi @cndoit18, Thank you very much for your contribution! It looks good. I have only one question and a suggestion.
> 
> Sorry for the delay, I was caught in other projects. 😞

Thank you very much, I hope to bring more contributions to this project and be able to speed up the progress of the roadmap. I would like to ask what are the requirements to be the core maintainer of this project?

File: pkg/sidecar/configs.go
Comment (cndoit18) [reply pkg/sidecar/configs.go:155]: Using `HasSuffix` will improve performance, using `getOrdinalFromHostname` will have better readability, which one do you think is better?

Comment (AMecea) [reply pkg/sidecar/configs.go:155]: I opt for readability, the performance gain is insignificant in this case

**Comment by AMecea** [comment]
Beside some minor comments all looks good. After you update the code I will merge this. Thanks again!

 > I would like to ask what are the requirements to be the core maintainer of this project?

We don't have yet requirements for maintainers. Please continue the good work, we would love to have more people working along us on this project. 😄

**Review by AMecea — APPROVED** [review]

**Comment by lizhongxuan** [comment]
Thanks!!!!

**Comment by cndoit18** [comment]
> Thanks!!!!

It is recommended that you turn on graceful scheduling for more stability

**Comment by lizhongxuan** [comment]
> > Thanks!!!!
> 
> It is recommended that you turn on graceful scheduling for more stability

update mysqlcluster  fail
```
2021-06-09T08:46:31.261593Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.20-11) starting as process 1
2021-06-09T08:46:31.273844Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2021-06-09T08:46:31.769019Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2021-06-09T08:46:32.079049Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: '/var/lib/mysql/mysqlx.sock' bind-address: '::' port: 33060
2021-06-09T08:46:32.200204Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2021-06-09T08:46:32.263329Z 0 [Warning] [MY-010604] [Repl] Neither --relay-log nor --relay-log-index were used; so replication may break when this MySQL server acts as a slave and has his hostname changed!! Please use '--relay-log=testww-mysql-0-relay-bin' to avoid this problem.
2021-06-09T08:46:32.266324Z 0 [Warning] [MY-010533] [Repl] Error during --relay-log-recovery: Could not locate rotate event from the master.
2021-06-09T08:46:32.266607Z 0 [Warning] [MY-013504] [Repl] Server was not able to find a rotate event from master server to initialize relay log recovery for channel ''. Skipping relay log recovery for the channel.
2021-06-09T08:46:32.302295Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.20-11'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 11, Revision 5b5a5d2.
2021-06-09T08:46:54.305007Z 30 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''. New state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
2021-06-09T08:46:54.311761Z 32 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
2021-06-09T08:46:54.315089Z 32 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@testww-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
2021-06-09T08:46:54.319253Z 32 [ERROR] [MY-010557] [Repl] Error reading packet from server for channel '': Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been (server_errno=1236)
2021-06-09T08:46:54.319292Z 32 [ERROR] [MY-013114] [Repl] Slave I/O for channel '': Got fatal error 1236 from master when reading data from binary log: 'Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been', Error_code: MY-013114
```

**Comment by lizhongxuan** [comment]
When update MySQL, I want to set master to read-only. How about this?

**Comment by cndoit18** [comment]
> > > Thanks!!!!
> > 
> > 
> > It is recommended that you turn on graceful scheduling for more stability
> 
> update mysqlcluster fail
> 
> ```
> 2021-06-09T08:46:31.261593Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.20-11) starting as process 1
> 2021-06-09T08:46:31.273844Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
> 2021-06-09T08:46:31.769019Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
> 2021-06-09T08:46:32.079049Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: '/var/lib/mysql/mysqlx.sock' bind-address: '::' port: 33060
> 2021-06-09T08:46:32.200204Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
> 2021-06-09T08:46:32.263329Z 0 [Warning] [MY-010604] [Repl] Neither --relay-log nor --relay-log-index were used; so replication may break when this MySQL server acts as a slave and has his hostname changed!! Please use '--relay-log=testww-mysql-0-relay-bin' to avoid this problem.
> 2021-06-09T08:46:32.266324Z 0 [Warning] [MY-010533] [Repl] Error during --relay-log-recovery: Could not locate rotate event from the master.
> 2021-06-09T08:46:32.266607Z 0 [Warning] [MY-013504] [Repl] Server was not able to find a rotate event from master server to initialize relay log recovery for channel ''. Skipping relay log recovery for the channel.
> 2021-06-09T08:46:32.302295Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.20-11'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 11, Revision 5b5a5d2.
> 2021-06-09T08:46:54.305007Z 30 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''. New state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
> 2021-06-09T08:46:54.311761Z 32 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
> 2021-06-09T08:46:54.315089Z 32 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@testww-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
> 2021-06-09T08:46:54.319253Z 32 [ERROR] [MY-010557] [Repl] Error reading packet from server for channel '': Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been (server_errno=1236)
> 2021-06-09T08:46:54.319292Z 32 [ERROR] [MY-013114] [Repl] Slave I/O for channel '': Got fatal error 1236 from master when reading data from binary log: 'Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been', Error_code: MY-013114
> ```

Did you create a new mysqlcluster? If it's old, you should have problems with dirty data, do you have slack? We can use it to chat

**Comment by lizhongxuan** [comment]
> > > > Thanks!!!!
> > > 
> > > 
> > > It is recommended that you turn on graceful scheduling for more stability
> > 
> > 
> > update mysqlcluster fail
> > ```
> > 2021-06-09T08:46:31.261593Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.20-11) starting as process 1
> > 2021-06-09T08:46:31.273844Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
> > 2021-06-09T08:46:31.769019Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
> > 2021-06-09T08:46:32.079049Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: '/var/lib/mysql/mysqlx.sock' bind-address: '::' port: 33060
> > 2021-06-09T08:46:32.200204Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
> > 2021-06-09T08:46:32.263329Z 0 [Warning] [MY-010604] [Repl] Neither --relay-log nor --relay-log-index were used; so replication may break when this MySQL server acts as a slave and has his hostname changed!! Please use '--relay-log=testww-mysql-0-relay-bin' to avoid this problem.
> > 2021-06-09T08:46:32.266324Z 0 [Warning] [MY-010533] [Repl] Error during --relay-log-recovery: Could not locate rotate event from the master.
> > 2021-06-09T08:46:32.266607Z 0 [Warning] [MY-013504] [Repl] Server was not able to find a rotate event from master server to initialize relay log recovery for channel ''. Skipping relay log recovery for the channel.
> > 2021-06-09T08:46:32.302295Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.20-11'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 11, Revision 5b5a5d2.
> > 2021-06-09T08:46:54.305007Z 30 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''. New state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
> > 2021-06-09T08:46:54.311761Z 32 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
> > 2021-06-09T08:46:54.315089Z 32 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@testww-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
> > 2021-06-09T08:46:54.319253Z 32 [ERROR] [MY-010557] [Repl] Error reading packet from server for channel '': Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been (server_errno=1236)
> > 2021-06-09T08:46:54.319292Z 32 [ERROR] [MY-013114] [Repl] Slave I/O for channel '': Got fatal error 1236 from master when reading data from binary log: 'Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been', Error_code: MY-013114
> > ```
> 
> Did you create a new mysqlcluster? If it's old, you should have problems with dirty data, do you have slack? We can use it to chat

new mysqlcluster.  wechart ?

**Comment by cndoit18** [comment]
> > > > > Thanks!!!!
> > > > 
> > > > 
> > > > It is recommended that you turn on graceful scheduling for more stability
> > > 
> > > 
> > > update mysqlcluster fail
> > > ```
> > > 2021-06-09T08:46:31.261593Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.20-11) starting as process 1
> > > 2021-06-09T08:46:31.273844Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
> > > 2021-06-09T08:46:31.769019Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
> > > 2021-06-09T08:46:32.079049Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: '/var/lib/mysql/mysqlx.sock' bind-address: '::' port: 33060
> > > 2021-06-09T08:46:32.200204Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
> > > 2021-06-09T08:46:32.263329Z 0 [Warning] [MY-010604] [Repl] Neither --relay-log nor --relay-log-index were used; so replication may break when this MySQL server acts as a slave and has his hostname changed!! Please use '--relay-log=testww-mysql-0-relay-bin' to avoid this problem.
> > > 2021-06-09T08:46:32.266324Z 0 [Warning] [MY-010533] [Repl] Error during --relay-log-recovery: Could not locate rotate event from the master.
> > > 2021-06-09T08:46:32.266607Z 0 [Warning] [MY-013504] [Repl] Server was not able to find a rotate event from master server to initialize relay log recovery for channel ''. Skipping relay log recovery for the channel.
> > > 2021-06-09T08:46:32.302295Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.20-11'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 11, Revision 5b5a5d2.
> > > 2021-06-09T08:46:54.305007Z 30 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''. New state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
> > > 2021-06-09T08:46:54.311761Z 32 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
> > > 2021-06-09T08:46:54.315089Z 32 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@testww-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
> > > 2021-06-09T08:46:54.319253Z 32 [ERROR] [MY-010557] [Repl] Error reading packet from server for channel '': Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been (server_errno=1236)
> > > 2021-06-09T08:46:54.319292Z 32 [ERROR] [MY-013114] [Repl] Slave I/O for channel '': Got fatal error 1236 from master when reading data from binary log: 'Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been', Error_code: MY-013114
> > > ```
> > 
> > 
> > Did you create a new mysqlcluster? If it's old, you should have problems with dirty data, do you have slack? We can use it to chat
> 
> new mysqlcluster. wechart ?



> > > > > Thanks!!!!
> > > > 
> > > > 
> > > > It is recommended that you turn on graceful scheduling for more stability
> > > 
> > > 
> > > update mysqlcluster fail
> > > ```
> > > 2021-06-09T08:46:31.261593Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.20-11) starting as process 1
> > > 2021-06-09T08:46:31.273844Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
> > > 2021-06-09T08:46:31.769019Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
> > > 2021-06-09T08:46:32.079049Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Socket: '/var/lib/mysql/mysqlx.sock' bind-address: '::' port: 33060
> > > 2021-06-09T08:46:32.200204Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
> > > 2021-06-09T08:46:32.263329Z 0 [Warning] [MY-010604] [Repl] Neither --relay-log nor --relay-log-index were used; so replication may break when this MySQL server acts as a slave and has his hostname changed!! Please use '--relay-log=testww-mysql-0-relay-bin' to avoid this problem.
> > > 2021-06-09T08:46:32.266324Z 0 [Warning] [MY-010533] [Repl] Error during --relay-log-recovery: Could not locate rotate event from the master.
> > > 2021-06-09T08:46:32.266607Z 0 [Warning] [MY-013504] [Repl] Server was not able to find a rotate event from master server to initialize relay log recovery for channel ''. Skipping relay log recovery for the channel.
> > > 2021-06-09T08:46:32.302295Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.20-11'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 11, Revision 5b5a5d2.
> > > 2021-06-09T08:46:54.305007Z 30 [System] [MY-010597] [Repl] 'CHANGE MASTER TO FOR CHANNEL '' executed'. Previous state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''. New state master_host='testww-mysql-1.mysql.kme-mysql-10272', master_port= 3306, master_log_file='', master_log_pos= 4, master_bind=''.
> > > 2021-06-09T08:46:54.311761Z 32 [Warning] [MY-010897] [Repl] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
> > > 2021-06-09T08:46:54.315089Z 32 [System] [MY-010562] [Repl] Slave I/O thread for channel '': connected to master 'sys_replication@testww-mysql-1.mysql.kme-mysql-10272:3306',replication started in log 'FIRST' at position 4
> > > 2021-06-09T08:46:54.319253Z 32 [ERROR] [MY-010557] [Repl] Error reading packet from server for channel '': Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been (server_errno=1236)
> > > 2021-06-09T08:46:54.319292Z 32 [ERROR] [MY-013114] [Repl] Slave I/O for channel '': Got fatal error 1236 from master when reading data from binary log: 'Slave has more GTIDs than the master has, using the master's SERVER_UUID. This may indicate that the end of the binary log was truncated or that the last binary log file was lost, e.g., after a power or disk failure when sync_binlog != 1. The master may or may not have rolled back transactions that were already replicated to the slave. Suggest to replicate any transactions that master has rolled back from slave to master, and/or commit empty transactions on master to account for transactions that have been', Error_code: MY-013114
> > > ```
> > 
> > 
> > Did you create a new mysqlcluster? If it's old, you should have problems with dirty data, do you have slack? We can use it to chat
> 
> new mysqlcluster. wechart ?

Ok, what is yours?

**Comment by lizhongxuan** [comment]
@cndoit18 wx234258202

**Comment by lizhongxuan** [comment]
Thanks @cndoit18  ,  I forgot to update the image of mysql-operator-sidecar , MySQLCluster is working properly.

**Pull request merged**
